src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -1152,18 +1159,19 @@ std::shared_ptr<ExpressionResult<Result>> ExpressionEvaluator::_evaluate_logical
 template <typename Result, typename Functor>
 std::shared_ptr<ExpressionResult<Result>> ExpressionEvaluator::_evaluate_binary_with_default_null_logic(
     const AbstractExpression& left_expression, const AbstractExpression& right_expression) {
-  auto result = std::shared_ptr<ExpressionResult<Result>>{};
+  std::vector<Result> values;
+  std::vector<bool> nulls;
```

Comment:
Just curious: Moving this out of the lambda really makes a difference, ct-wise?

Reply:
I didn't measure this specifically. Generally, when trying to keep the size of the lambda short in terms of instructions, I think it helps to also reduce the number of source code lines.